src/App.tsx

```diff
@@ -1,10 +1,49 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { ErrorMessage } from './types/ErrorMessage';
+import { getTodos, USER_ID } from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList.tsx';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoStatus, setTodoStatus] = useState<TodoStatus>(TodoStatus.ALL);
+  const [hasError, setHasError] = useState<ErrorMessage>(ErrorMessage.NONE);
+
+  const visibleTodos = useMemo(() => {
+    return (todos.filter((todo) => {
+      switch (todoStatus) {
+        case TodoStatus.ACTIVE:
+          return !todo.completed;
+
+        case TodoStatus.COMPLETED:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    }));
```

Comment:
No need to filter in case TodoStatus.ALL

Reply:
If I'm not mistaken, in this case I don't do anything, only Completed and Active